WeakAuras/RegionTypes/DynamicGroup.lua

```diff
@@ -813,7 +814,46 @@ local function modify(parent, region, data)
     WeakAuras.StartProfileSystem("dynamicgroup")
     WeakAuras.StartProfileAura(data.id)
     local newPositions = {}
-    self.growFunc(newPositions, self.sortedChildren)
+    local anchorPerUnit = data.anchorFrameType == "NAMEPLATE" or data.anchorFrameType == "UNITFRAME"
+    if data and anchorPerUnit then
+      local units = {}
+      -- build a list keyed by unit
+      for i, regionData in ipairs(self.sortedChildren) do
+        local anchor, unit
+        local state = regionData.region.state
+        if state and state.unit then
+          unit = state.unit
+          if data.anchorFrameType == "NAMEPLATE" then
+            anchor = C_NamePlate.GetNamePlateForUnit(unit)
+          elseif data.anchorFrameType == "UNITFRAME" then
+            anchor = GetUnitFrame(unit)
+          end
+        end
+        if anchor then
+          regionData.region.anchor = anchor
+          units[unit] = units[unit] or {}
```

Comment:
I think it's better to have the frames per anchor instead of per unit. That makes a difference if GetUnitFrame returns the same anchor for different units.

Reply:
changed to indesx by frame instead of unit